src/main/java/org/cyclonedx/util/ExternalReferenceSerializer.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * This file is part of CycloneDX Core (Java).
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright (c) Steve Springett. All Rights Reserved.
+ */
+package org.cyclonedx.util;
+
+import java.io.IOException;
+
+import javax.xml.stream.XMLStreamException;
+import javax.xml.stream.XMLStreamWriter;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.ser.std.StdSerializer;
+import com.fasterxml.jackson.dataformat.xml.ser.ToXmlGenerator;
+import org.cyclonedx.model.ExternalReference;
+
+public class ExternalReferenceSerializer extends StdSerializer<ExternalReference>
+{
+  public ExternalReferenceSerializer() {
+    this(null);
+  }
+
+  public ExternalReferenceSerializer(final Class t) {
+    super(t);
+  }
+
+  @Override
+  public void serialize(
+      final ExternalReference extRef, final JsonGenerator gen, final SerializerProvider provider) throws IOException
+  {
+    if (gen instanceof ToXmlGenerator) {
+      final ToXmlGenerator toXmlGenerator = (ToXmlGenerator) gen;
+      final XMLStreamWriter staxWriter = toXmlGenerator.getStaxWriter();
+      if (extRef.getType() != null && extRef.getUrl() != null && BomUtils.validateUrlString(extRef.getUrl())) {
```

Comment:
what about extracting this condition as a variable, it can be used in line 68 also

Reply:
A predicate could be created

Reply:
d4b9155